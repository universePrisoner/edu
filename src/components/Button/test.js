import React from "react";
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Button } from './index';

Enzyme.configure({ adapter: new Adapter() });

describe('Button', () => {
    let onClick = () => console.log(1);

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Button onClick={() => onClick('3')}>Give me more</Button>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('here is short snapshot', () => {
        const component = renderer.create(<Button onClick={() => onClick('3')}>Give me more</Button>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('show one button', () => {
        const element = shallow(<Button onClick={() => onClick('3')}>Give me more</Button>)

        expect(element.find('button[type="button"]').length).toBe(1);
    });
});