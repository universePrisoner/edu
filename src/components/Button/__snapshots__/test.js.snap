// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button here is short snapshot 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button"
>
  Give me more
</button>
`;
